python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -214,7 +214,9 @@ public static boolean isValidForProjectPythonVersion(List<String> validForPython
     if (validForPythonVersions.isEmpty()) {
       return true;
     }
-    if (supportedPythonVersions.stream().allMatch("312"::equals) && validForPythonVersions.contains("311")) {
+    // TODO: SONARPY-1522 - remove this workaround when we will have all the stubs for Python 3.12.
+    if (supportedPythonVersions.stream().allMatch(PythonVersionUtils.Version.V_312.serializedValue()::equals)
+      && validForPythonVersions.contains(PythonVersionUtils.Version.V_312.serializedValue())) {
```

Comment:
Changed the bug in version number to `PythonVersionUtils.Version.V_312` in 159351aadb39f43eaca139cc83b23b716b91769a.